pkg/cmd/cli/cmd/debug.go

```diff
@@ -531,17 +532,24 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*kap
 		if pod.Labels == nil {
 			pod.Labels = make(map[string]string)
 		}
-		pod.Labels[debugPodLabelName] = pod.Name
 	} else {
-		pod.Labels = map[string]string{debugPodLabelName: pod.Name}
+		pod.Labels = map[string]string{}
 	}
 	// always clear the NodeName
 	pod.Spec.NodeName = o.NodeName
 
 	pod.ResourceVersion = ""
 	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
 	// TODO: shorten segments, make incrementing?
-	pod.Name = fmt.Sprintf("%s-debug", pod.Name)
+	originalName := pod.Name
+	for i := len(originalName); i > 0; i-- {
```

Comment:
> Why do we need this behavior?

Colons are required in istag names and disallowed in pod names.  This decimates names backwards until they meeting length requirements (problem today) and don't have illegal characters (problem with istags).